src/Voron/Data/Lookups/Lookup.cs

```diff
@@ -407,7 +407,10 @@ private void RemoveEntryFromPage(ref CursorState state, ref TLookupKey key, int
         Debug.Assert(state.Header->Upper - state.Header->Lower >= 0);
         Debug.Assert(state.Header->FreeSpace <= Constants.Storage.PageSize - PageHeader.SizeOf);
         Debug.Assert(k == key.ToLong(), "k == key.ToLong()");
-        key.OnKeyRemoval(this);
+        
+        if (state.Header->IsLeaf)
```

Comment:
Not sure if that is a problem, but that will leak a key.

Scenario:


```
-Root:
   - 10 (branch)
   - - 10
   - - 15
   - - 19
   - 20 (branch)
   -  - 20
   - - 24
   - - 27
   - 30
```

Now, delete `10`

Structure is now:

```
-Root:
   - 10 (branch)
   - - 11 * note, we deleted 10 as the key here, but it is still the one above)
   - - 15
   - - 19
   - 20 (branch)
   -  - 20
   - - 24
   - - 27
   - 30
```

Reply:
Good catch, fixed now.